lib/dal/src/component/debug.rs

```
@@ -0,0 +1,177 @@
+        };
+
+        Ok(debug_view)
+    }
+    #[instrument(level = "info", skip_all)]
+    pub async fn get_attribute_value_tree_for_component(
+        ctx: &DalContext,
+        component_id: ComponentId,
+    ) -> ComponentDebugViewResult<HashMap<AttributeValueId, Vec<AttributeValueId>>> {
+        let mut child_values = HashMap::new();
+        // Get the root attribute value and load it into the work queue.
+        let root_attribute_value_id = Component::root_attribute_value_id(ctx, component_id).await?;
+        let workspace_snapshot = ctx.workspace_snapshot()?;
+
+        let mut work_queue = VecDeque::from([root_attribute_value_id]);
+        while let Some(attribute_value_id) = work_queue.pop_front() {
+            // Collect all child attribute values.
+            let mut cache: Vec<(AttributeValueId, Option<String>)> = Vec::new();
+            {
+                let mut child_attribute_values_with_keys_by_id: HashMap<
+                    AttributeValueId,
+                    (NodeIndex, Option<String>),
+                > = HashMap::new();
+
+                for (edge_weight, _, target_idx) in workspace_snapshot
+                    .edges_directed(attribute_value_id, Direction::Outgoing)
+                    .await?
+                {
+                    if let EdgeWeightKind::Contain(key) = edge_weight.kind() {
+                        let child_id = workspace_snapshot
+                            .get_node_weight(target_idx)
+                            .await?
+                            .id()
+                            .into();
+                        child_attribute_values_with_keys_by_id
+                            .insert(child_id, (target_idx, key.to_owned()));
+                    }
+                }
+
+                let maybe_ordering =
+                    AttributeValue::get_child_av_ids_for_ordered_parent(ctx, attribute_value_id)
+                        .await
+                        .ok();
```

Are we eating a potential error here?